associated-token-account/program/src/instruction.rs

```diff
@@ -34,6 +34,20 @@ pub enum AssociatedTokenAccountInstruction {
     ///   4. `[]` System program
     ///   5. `[]` SPL Token program
     CreateIdempotent,
+    /// Transfers from and closes a nested associated token account: an
+    /// associated token account owned by an associated token account.
+    ///
+    /// The tokens are moved from the nested associated token account to the
+    /// wallet's associated token account.
+    ///
+    ///   0. `[writeable]` Nested associated token account, must be owned by `3`
+    ///   1. `[]` Token mint for the nested associated token account.
+    ///   2. `[writeable]` Wallet's associated token account.
+    ///   3. `[]` Owner associated token account address, must be owned by `5`
+    ///   4. `[]` Token mint for the owner associated token account
+    ///   5. `[writeable, signer]` Wallet address for the owner associated token account
+    ///   6. `[]` SPL Token program
+    CloseNested,
```

Comment:
bikeshed: related to https://github.com/solana-labs/solana-program-library/pull/2889/files#r827912548, i somewhat want to avoid the `Close` verb (i know this is borrowed from `Token::CloseAccount` for consistency).

how about `RescueAndCloseNested` (mouthful naming indicates something wary;) or `RescueNested`

Reply:
That's a good point, also considering it's not just closing the account.  I was going to suggest `RecoverAndCloseNested` as an alternative, but I like that `Rescue` includes the connotation that something bad / wrong has happened. @CriesofCarrots thoughts?

Reply:
Fair point. I'm in favor of removing the word close altogether, and my preference is `RecoverNested`

Reply:
Alright, going with `RecoverNested`!

Reply:
I think I'd prefer something more derogatory than "nested."  It doesn't quite have the "that you are here implies that you have done something very stupid and should be ashamed of yourself," connotation that this situation deserves

Reply:
Any suggestions for words that convey that? `Inappropriate`? `Improper`? `Perverse`?  `RecoverInappropriateNested`?

Reply:
`RecoverNested` seems fine to me. The client tool that wraps this instruction can alway hurl insults

Reply:
my :bike: is parked under `RecoverNested` from the first impression of the ix name with fresh brain.

Reply:
`RecoverNested` it is!